time measurements changes

busmangit opened the pull request:
time measurements changes in algorithms implementations. Mainly in run funcionts

busmangit committed:
time measurements

time measurements for benchmarking (printf), some include added and required local scope functions

busmangit committed:
Test simple change

Test simple change

busmangit committed:
Merge branch 'benchmarking' of https://github.com/ParticleTracking2/PTrack2 into benchmarking

juanssl commented on src/Algorithm/Implementations/Chi2Algorithm.cpp:
```diff
@@ -64,45 +67,84 @@ void Chi2Algorithm::setData(ParameterContainer *pc){
 	else
 		_data = MyImageFactory::makeRawImgFromFile(pc->getParamAsString("-i"));
+
+    printf("scanning string: %s\n", pc->getParamAsString("-i").c_str());
```
Use MyLogger::log()->debug instead of printf

juanssl reviewed (commented):
Too many changes in Chi2Algorithm.cpp, Chi2HDAlgorithm.cpp and Chi2HDCudaAlgorithm.cpp including opening a file. Consider create another Algorithm type instead to avoid affecting the performance of the original algorithm.

busmangit committed:
qhull library update for mac os sierra

2012 qhull library does not compile on mac os sierra, most recent 2015 version is needed and it is compatible

busmangit committed:
simplified makefile

single recursive makefile using pkg-config

busmangit committed:
macport update

for mac os 10.12 (sierra)

busmangit committed:
added include

for single makefile modification

busmangit committed:
benchmarking auxiliary files

octave files and sh scripts to make benchmarking

busmangit committed:
Time measurement modifications and simplified compilation

busmangit committed:
simplified installation

single apt-get install command and updated libraries names

busmangit committed:
added install part of new makefile

busmangit committed:
makefile simplification

now using single makefile

busmangit committed:
modified for ubuntu 16.04

modified for ubuntu 16.04 and single apt-get command

busmangit committed:
typo

typo correction

juanssl reviewed (commented):
Do we need this file? ptrack-gentest/octave-workspace

juanssl commented:
Branch for merging should be dev

juanssl commented:
Too many changes in the algorithms... Doesn't looks like it's going to make it to dev. Better keep it in your branch since all the changes are specific for benchmarking.

busmangit committed:
corrections in documentations

for benchmarking
